java-client/src/main/java/org/apache/pegasus/security/KerberosProtocol.java

```diff
@@ -50,19 +50,20 @@ class KerberosProtocol implements AuthProtocol {
   // request. The JAAS framework defines the term "subject" to represent the source of a request. A
   // subject may be any entity, such as a person or a service.
   private Subject subject;
-  private String serviceName;
-  private String serviceFqdn;
-  private String keyTab;
-  private String principal;
+  private final String serviceName;
+  private final String serviceFqdn;
+  private final String keyTab;
+  private final String principal;
   final int CHECK_TGT_INTEVAL_SECONDS = 10;
   final ScheduledExecutorService service =
       Executors.newSingleThreadScheduledExecutor(
           new ThreadFactory() {
             @Override
             public Thread newThread(Runnable r) {
-              Thread t = new Thread(r);
+              Thread t = new Thread(r, "TGT renew for pegasus");
```

Comment:
Better introduce a unique suffix or prefix, otherwise if there are multiple client you will see multiple threads with the same name.